lib/engine/step/g_18_al/train.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require_relative '../single_depot_train_buy_before_phase4'
+
+module Engine
+  module Step
+    module G18AL
+      class Train < SingleDepotTrainBuyBeforePhase4
+        def process_buy_train(action)
+          ability = action.ability
+
+          super
+
+          return if ability.nil?
```

Comment:
super

action.entity.ability(:train) do |ability|
 ...

Reply:
Fixed, and this part was rewritten as well. Now the use! is done in the g_18_al/train.rb code.